tests/shouldwork/BitVector/UnpackUndefined.hs

```diff
@@ -4,4 +4,4 @@ import Clash.Sized.Internal.BitVector
 
 -- https://github.com/clash-lang/clash-compiler/issues/804
 state = unpack undefined#    :: Vec 3 Bit
-topEntity = register @System state
+topEntity a = register @System state a
```

Comment:
I don't understand why this needs to change. Is this explained anywhere?

Reply:
It's explained somewhere on the internet. The monomorphism restriction is annoying.

Reply:
https://wiki.haskell.org/Monomorphism_restriction